qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -1668,6 +1680,18 @@ def _on_select_client_certificate(self, selection):
         else:
             selection.selectNone()
 
+    def _on_recommended_state_changed(
```

Comment:
```suggestion
    @pyqtSlot(QWebEnginePage.LifecycleState)
    def _on_recommended_state_changed(
```

to avoid PyQt taking a detour via Python, which also might fix issues with the slot getting called on deleted tabs.